src/storage/store/account/event.rs

```diff
@@ -211,3 +232,455 @@ impl HubEvent {
+        let event_id = generator.generate_id(&block_confirmed).unwrap();
+
+        // Verify the event ID is correctly constructed
+        let expected_id = HubEventIdGenerator::make_event_id(456, 0);
+        assert_eq!(event_id, expected_id);
+
+        // Test extraction works correctly
+        let (height, sequence) = HubEventIdGenerator::extract_height_and_seq(event_id);
+        assert_eq!(height, 456);
+        assert_eq!(sequence, 0);
+
+        // Test regular event ID generation
+        let regular_event = create_test_event(HubEventType::MergeMessage);
+        let event_id = generator.generate_id(&regular_event).unwrap();
+
+        let expected_id = HubEventIdGenerator::make_event_id(456, 1);
+        assert_eq!(event_id, expected_id);
+
+        let (height, sequence) = HubEventIdGenerator::extract_height_and_seq(event_id);
+        assert_eq!(height, 456);
+        assert_eq!(sequence, 1);
+    }
+
+    #[test]
+    fn test_event_id_generator_state_management() {
+        let mut generator = HubEventIdGenerator::new(Some(100), Some(5));
+
+        // Test initial state
+        assert_eq!(generator.current_height, 100);
+        assert_eq!(generator.current_seq, 5);
+
+        // Test height transition
+        generator.set_current_height(200);
+        assert_eq!(generator.current_height, 200);
+        assert_eq!(generator.current_seq, 0); // Should reset to 0
+
+        // Test sequence progression
+        let regular_event = create_test_event(HubEventType::MergeMessage);
+        let event_id = generator.generate_id(&regular_event).unwrap();
+        let (height, sequence) = HubEventIdGenerator::extract_height_and_seq(event_id);
+
+        assert_eq!(height, 200);
+        assert_eq!(sequence, 1);
+        assert_eq!(generator.current_seq, 2); // Should be incremented
+
+        // Test BLOCK_CONFIRMED doesn't affect sequence counter
+        let block_confirmed = create_test_event(HubEventType::BlockConfirmed);
+        let event_id = generator.generate_id(&block_confirmed).unwrap();
+        let (height, sequence) = HubEventIdGenerator::extract_height_and_seq(event_id);
+
+        assert_eq!(height, 200);
+        assert_eq!(sequence, 0);
+        assert_eq!(generator.current_seq, 2); // Should remain unchanged
+    }
+
+    #[test]
+    fn test_make_event_id_and_extraction() {
+        // Test basic event ID creation
+        let height = 12345;
+        let sequence = 678;
+        let event_id = HubEventIdGenerator::make_event_id(height, sequence);
+
+        let (extracted_height, extracted_sequence) =
+            HubEventIdGenerator::extract_height_and_seq(event_id);
+        assert_eq!(extracted_height, height);
+        assert_eq!(extracted_sequence, sequence);
+
+        // Test with sequence that needs masking
+        let height = 999;
+        let sequence = 2u64.pow(SEQUENCE_BITS) + 100; // Should be masked
+        let event_id = HubEventIdGenerator::make_event_id(height, sequence);
```

Comment:
Yeah, again this makes me question whether `make_event_id` should be public. In the `generate_id` function we have guardrails so that we error on sequence numbers which are too high.